kmain.py

```diff
@@ -151,127 +219,6 @@ def categorical_accuracy_mod(y_true, y_pred):
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-  np.save("out_9_lrp", analysis)
-
-  print("New model 3")
-  
-  new_model_1 = Model(model.inputs, model.layers[-9].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-  np.save("out_8_lrp", analysis)
-
-  print("New model 4")
-  
-  new_model_1 = Model(model.inputs, model.layers[-12].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-  np.save("out_7_lrp", analysis)
-
-  print("New model 5")
-  
-  new_model_1 = Model(model.inputs, model.layers[-15].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-  np.save("out_6_lrp", analysis)
-
-  print("New model 6")
-  
-  new_model_1 = Model(model.inputs, model.layers[-18].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-  np.save("out_5_lrp", analysis)
-
-  
-  print("New model 7")
-  
-  new_model_1 = Model(model.inputs, model.layers[-21].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-
-  np.save("out_4_lrp", analysis)
-
-  print("New model 8")
-  
-  new_model_1 = Model(model.inputs, model.layers[-24].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-
-  np.save("out_3_lrp", analysis)
-
-  print("New model 9")
-  
-  new_model_1 = Model(model.inputs, model.layers[-27].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-
-  np.save("out_2_lrp", analysis)
-
-  print("New model 10")
-  
-  new_model_1 = Model(model.inputs, model.layers[-30].output)
-  new_model_1.set_weights(model.get_weights())
-  new_model_1.summary()
-
-  analyzer = innvestigate.create_analyzer("lrp.z", new_model_1)
-  analysis = analyzer.analyze(x_train)
-  print("analysis: "+str(analysis))
-
-  np.save("out_1_lrp", analysis)
```

Comment:
Hi @thevoxium! Thanks a lot for the PR. May I know why you have removed these lines?

Reply:
I was just experimenting with that can I implement it using functions. I will change it in the next commit.

Reply:
Yes, it would be great if you can implement it using a function. Moreover, you could perhaps make a list containing the no of layers you wish to have, and have a loop iterating over that which keeps calling the LRP function.

Reply:
@americast I asked a doubt regarding LRP on gitter. It would be great if you look at please.